lib/components/form/batch-preferences.tsx

```diff
@@ -40,12 +43,21 @@ class BatchPreferences extends Component<{
    * Typescript TODO: combinations and queryParams need types
    */
   onQueryParamChange = (newQueryParams: any) => {
-    const { config, query, setQueryParam } = this.props
-    const disabledModes = query.disabledModes || []
+    const { config, modeOptions, query, setQueryParam } = this.props
+    const enabledModes = query.enabledModes || modeOptions
     const combinations = config.modes.combinations
-      .filter((combination: Combination) => {
-        const modesInCombination = combination.mode.split(',')
-        return !modesInCombination.find((m) => disabledModes.includes(m))
+      .filter((c: Combination) => {
+        if (c.requiredModes) {
+          return c.requiredModes.every((m) => enabledModes.includes(m))
+        } else {
+          // This is for backwards compatibility
+          // In case a combination does not include requiredModes.
+          console.warn(
+            `Combination ${c.mode} does not have any specified required modes.`
+          )
+          const modesInCombination = c.mode.split(',')
+          return modesInCombination.every((m) => enabledModes.includes(m))
```

Comment:
Would it be possible to extract this back into the util file that you deleted those old methods from? Let me know if that didn't make sense